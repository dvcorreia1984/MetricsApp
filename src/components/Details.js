import React from 'react';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import PropTypes from 'prop-types';

const Card = ({ label, value }) => (
  <li className="flex text-2xl h-24 p-5 items-center text-white gap-5 even:bg-[#35548b]">
    <div className="w-7/12 pl-5">{label}</div>
    <div className="w-4/12">{value}</div>
  </li>
);

const Details = () => {
  const params = useParams();
  const { subregion } = params;
  const countries = useSelector((state) => state.countries.countries);

  const selectedCountry = countries.find((country) => country.name === subregion);

  return (
    <div>
      <div className="flex h-40 bg-[#5487e3] justify-center gap-10">
        <div className="flex w-1/4 text-blue-950 py-3 items-center">
          <div>
            <img src={selectedCountry?.flag} alt={subregion} className="w-40 h-30 rounded-lg" />
          </div>
        </div>
        <div className="flex flex-col justify-center items-start text-white">
          <h1 className="font-bold">{subregion?.toUpperCase()}</h1>
          <h3 className="mt-[-5px] text-[10px]">{`Population: ${selectedCountry?.population}`}</h3>
        </div>
      </div>
      <div className="bg-[#35548b] pl-5 h-8 text-white text-sm">STATS BY COUNTRY</div>
      <div>
        <ul>
          <Card label="CAPITAL" value={selectedCountry?.capital} />
          <Card label="POPULATION" value={selectedCountry?.population} />
          <Card label="AREA" value={selectedCountry?.area} />
          <Card label="CURRENCY" value={selectedCountry?.currencies[0]?.code} />
          <Card label="CALLING CODE" value={selectedCountry?.callingCodes} />
          <Card label="TIMEZONES" value={selectedCountry?.timezones} />
        </ul>
      </div>
    </div>
  );
};

Card.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
};

export default Details;
