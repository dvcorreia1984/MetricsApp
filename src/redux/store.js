// store.js
import { configureStore } from '@reduxjs/toolkit';
import countryReducer from './country/countrySlice';

export const store = configureStore({
  reducer: {
    countries: countryReducer,
  },
});
