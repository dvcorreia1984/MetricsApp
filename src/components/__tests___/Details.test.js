import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import Details from '../Details'; // Import your Details component

describe('Details Component', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        {' '}
        <Details />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
