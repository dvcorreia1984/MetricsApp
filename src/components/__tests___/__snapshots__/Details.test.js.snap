// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component should match the snapshot 1`] = `
Array [
  " ",
  <div>
    <div
      className="flex h-40 bg-[#5487e3] justify-center gap-10"
    >
      <div
        className="flex w-1/4 text-blue-950 py-3 items-center"
      >
        <div>
          <img
            className="w-40 h-30 rounded-lg"
          />
        </div>
      </div>
      <div
        className="flex flex-col justify-center items-start text-white"
      >
        <h1
          className="font-bold"
        />
        <h3
          className="mt-[-5px] text-[10px]"
        >
          Population: undefined
        </h3>
      </div>
    </div>
    <div
      className="bg-[#35548b] pl-5 h-8 text-white text-sm"
    >
      STATS BY COUNTRY
    </div>
    <div>
      <ul>
        <li
          className="flex text-2xl h-24 p-5 items-center text-white gap-5 even:bg-[#35548b]"
        >
          <div
            className="w-7/12 pl-5"
          >
            CAPITAL
          </div>
          <div
            className="w-4/12"
          />
        </li>
        <li
          className="flex text-2xl h-24 p-5 items-center text-white gap-5 even:bg-[#35548b]"
        >
          <div
            className="w-7/12 pl-5"
          >
            POPULATION
          </div>
          <div
            className="w-4/12"
          />
        </li>
        <li
          className="flex text-2xl h-24 p-5 items-center text-white gap-5 even:bg-[#35548b]"
        >
          <div
            className="w-7/12 pl-5"
          >
            AREA
          </div>
          <div
            className="w-4/12"
          />
        </li>
        <li
          className="flex text-2xl h-24 p-5 items-center text-white gap-5 even:bg-[#35548b]"
        >
          <div
            className="w-7/12 pl-5"
          >
            CURRENCY
          </div>
          <div
            className="w-4/12"
          />
        </li>
        <li
          className="flex text-2xl h-24 p-5 items-center text-white gap-5 even:bg-[#35548b]"
        >
          <div
            className="w-7/12 pl-5"
          >
            CALLING CODE
          </div>
          <div
            className="w-4/12"
          />
        </li>
        <li
          className="flex text-2xl h-24 p-5 items-center text-white gap-5 even:bg-[#35548b]"
        >
          <div
            className="w-7/12 pl-5"
          >
            TIMEZONES
          </div>
          <div
            className="w-4/12"
          />
        </li>
      </ul>
    </div>
  </div>,
]
`;
